src/components/search/results.tsx

```diff
@@ -54,15 +80,105 @@ const Results: React.FunctionComponent<ResultsProps> = React.memo(
-                ) : null}
+                )}
+
+                {!isValidating && (
+                    <>
+                        {error && (
+                            <div>Something went wrong, please try again</div>
+                        )}
+                        {!error && (
+                            <>
+                                {!!resultsToShow.length && (
+                                    // Needs to be wrapped in a div because Safari isn't the best with grid...
+                                    // https://stackoverflow.com/questions/44770074/css-grid-row-height-safari-bug
+                                    <div>
+                                        <div
+                                            data-testid="search-results"
+                                            sx={dataStyles(layout)}
+                                        >
+                                            {resultsToShow.map(item => (
+                                                <Card
+                                                    key={item.slug}
+                                                    sx={extraCardStyles}
+                                                    hideTagsOnMobile={
+                                                        layout === 'list'
+                                                    }
+                                                    {...getCardProps(
+                                                        item,
+                                                        layout === 'list'
+                                                            ? 'list'
+                                                            : 'medium'
+                                                    )}
+                                                />
+                                            ))}
+                                        </div>
+
+                                        {showLoadMoreButton && (
+                                            <div
+                                                sx={{
+                                                    display: 'flex',
+                                                    justifyContent: 'center',
+                                                    marginTop: [
+                                                        'inc70',
+                                                        null,
+                                                        'inc90',
+                                                    ],
+                                                }}
+                                            >
+                                                {!isValidating && results && (
+                                                    <a
+                                                        href={loadMoreHref}
+                                                        onClick={onLoadMore}
```

Comment:
I don't think this is behaving the way we want it to on every page.

An example of what's broken: When I go to http://localhost:3000/developer/languages/javascript/code-examples/ and click load more, it is going to an entirely different page http://localhost:3000/developer/languages/javascript?page=2.

I can't tell yet what the cause is, but wondering if it has to do with the memoization?